pom.xml

```diff
@@ -112,80 +112,80 @@
+    <mariadb.version>3.4.1</mariadb.version>
     <maven-assembly.version>3.7.1</maven-assembly.version>
-    <maven-clean.version>3.3.2</maven-clean.version>
+    <maven-clean.version>3.4.0</maven-clean.version>
     <maven-compiler.version>3.13.0</maven-compiler.version>
-    <maven-deploy.version>3.1.1</maven-deploy.version>
-    <maven-download.version>1.9.0</maven-download.version>
+    <maven-deploy.version>3.1.3</maven-deploy.version>
+    <maven-download.version>1.11.0</maven-download.version>
     <maven-eclipse.version>2.10</maven-eclipse.version>
-    <maven-enforcer.version>3.4.1</maven-enforcer.version>
-    <maven-gpg.version>3.2.2</maven-gpg.version>
-    <maven-install.version>3.1.1</maven-install.version>
+    <maven-enforcer.version>3.5.0</maven-enforcer.version>
+    <maven-gpg.version>3.2.7</maven-gpg.version>
+    <maven-install.version>3.1.3</maven-install.version>
     <maven-jacoco.version>0.8.12</maven-jacoco.version>
-    <maven-jar.version>3.3.0</maven-jar.version>
-    <maven-javadoc.version>3.6.3</maven-javadoc.version>
-    <maven-jxr.version>3.3.2</maven-jxr.version>
-    <maven-pmd.version>3.21.2</maven-pmd.version>
-    <maven-project-info.version>3.5.0</maven-project-info.version>
-    <maven-release.version>3.0.1</maven-release.version>
+    <maven-jar.version>3.4.2</maven-jar.version>
+    <maven-javadoc.version>3.10.1</maven-javadoc.version>
+    <maven-jxr.version>3.5.0</maven-jxr.version>
+    <maven-pmd.version>3.25.0</maven-pmd.version>
+    <maven-project-info.version>3.7.0</maven-project-info.version>
+    <maven-release.version>3.1.1</maven-release.version>
     <maven-resources.version>3.3.1</maven-resources.version>
-    <maven-shade.version>3.5.2</maven-shade.version>
-    <maven-site.version>4.0.0-M13</maven-site.version>
+    <maven-shade.version>3.6.0</maven-shade.version>
+    <maven-site.version>4.0.0-M16</maven-site.version>
     <maven-source.version>3.3.1</maven-source.version>
-    <maven-surefire.version>3.2.5</maven-surefire.version>
-    <metrics.version>4.2.25</metrics.version>
-    <mockito.version>5.11.0</mockito.version>
-    <mssql.version>12.6.1.jre11</mssql.version>
-    <mysql.version>8.3.0</mysql.version>
-    <nexus-staging-maven.version>1.6.13</nexus-staging-maven.version>
+    <maven-surefire.version>3.5.1</maven-surefire.version>
+    <metrics.version>4.2.28</metrics.version>
+    <mockito.version>5.14.2</mockito.version>
+    <mssql.version>12.8.1.jre11</mssql.version>
+    <mysql.version>9.1.0</mysql.version>
+    <nexus-staging-maven.version>1.7.0</nexus-staging-maven.version>
     <node.version>v20.12.1</node.version> <!-- https://nodejs.org/en/download/releases/ -->
     <npm.version>10.5.0</npm.version>     <!-- https://github.com/npm/cli/releases -->
     <nitor-matchers.version>1.3</nitor-matchers.version>
-    <ojdbc.version>23.2.0.0</ojdbc.version>
-    <postgresql.version>42.7.3</postgresql.version>
+    <ojdbc.version>23.5.0.24.07</ojdbc.version>
```

Comment:
The Oracle integration test failed for a real reason. Every ojdbc-version >23.2.0.0 gives me the following in the Oracle integration tests.

```
    <system-err><![CDATA[2024-10-24 12:59:11,723 ERROR [nflow-executor-2] xecutor.WorkflowStateProcessor - [31] [] Failed to save workflow instance 31 new state, retrying after PT60S.
org.springframework.jdbc.UncategorizedSQLException: PreparedStatementCallback; uncategorized SQLException for SQL [update nflow_workflow set status = ?, state = ?, state_text = ?, next_activation = (case when ? is null then null when external_next_activation is null then ? else least(?, external_next_activation) end), external_next_activation = null, executor_id = ?, retries = ?, business_key = ?, started = (case when started is null then ? else started end) where id = ? and executor_id = 9]; SQL state [99999]; error code [17004]; ORA-17004: Invalid column type: 268435455
https://docs.oracle.com/error-help/db/ora-17004/
        at org.springframework.jdbc.core.JdbcTemplate.translateException(JdbcTemplate.java:1573)
        at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:667)
        at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:960)
        at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:1015)
        at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:1025)
        at io.nflow.engine.internal.dao.WorkflowInstanceDao.updateWorkflowInstance(WorkflowInstanceDao.java:357)
        at io.nflow.engine.internal.dao.WorkflowInstanceDao$6.doInTransactionWithoutResult(WorkflowInstanceDao.java:375)
        at org.springframework.transaction.support.TransactionCallbackWithoutResult.doInTransaction(TransactionCallbackWithoutResult.java:36)
        at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:140)
        at io.nflow.engine.internal.dao.WorkflowInstanceDao.updateWorkflowInstanceWithTransaction(WorkflowInstanceDao.java:372)
        at io.nflow.engine.internal.dao.WorkflowInstanceDao.updateWorkflowInstanceAfterExecution(WorkflowInstanceDao.java:347)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:343)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:699)
        at io.nflow.engine.internal.dao.WorkflowInstanceDao$$SpringCGLIB$$0.updateWorkflowInstanceAfterExecution(<generated>)
        at io.nflow.engine.internal.executor.WorkflowStateProcessor.persistWorkflowInstanceState(WorkflowStateProcessor.java:341)
        at io.nflow.engine.internal.executor.WorkflowStateProcessor.saveWorkflowInstanceState(WorkflowStateProcessor.java:309)
        at io.nflow.engine.internal.executor.WorkflowStateProcessor.runImpl(WorkflowStateProcessor.java:211)
        at io.nflow.engine.internal.executor.WorkflowStateProcessor.run(WorkflowStateProcessor.java:130)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.sql.SQLException: ORA-17004: Invalid column type: 268435455
https://docs.oracle.com/error-help/db/ora-17004/
        at oracle.jdbc.driver.OracleStatement.getInternalType(OracleStatement.java:5233)
        at oracle.jdbc.driver.OraclePreparedStatement.setNullCritical(OraclePreparedStatement.java:4898)
        at oracle.jdbc.driver.OraclePreparedStatement.setNull(OraclePreparedStatement.java:4881)
        at oracle.jdbc.driver.OraclePreparedStatementWrapper.setNull(OraclePreparedStatementWrapper.java:942)
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.setNull(HikariProxyPreparedStatement.java)
        at org.springframework.jdbc.core.StatementCreatorUtils.setNull(StatementCreatorUtils.java:287)
        at org.springframework.jdbc.core.StatementCreatorUtils.setParameterValueInternal(StatementCreatorUtils.java:238)
        at org.springframework.jdbc.core.StatementCreatorUtils.setParameterValue(StatementCreatorUtils.java:173)
        at org.springframework.jdbc.core.ArgumentPreparedStatementSetter.doSetValue(ArgumentPreparedStatementSetter.java:72)
        at org.springframework.jdbc.core.ArgumentPreparedStatementSetter.setValues(ArgumentPreparedStatementSetter.java:51)
        at org.springframework.jdbc.core.JdbcTemplate.lambda$update$2(JdbcTemplate.java:963)
        at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:648)
        ... 21 common frames omitted
]]></system-err>
```